LibraBFT/Impl/Consensus/BlockStorage/BlockStore.agda

```diff
@@ -34,6 +48,44 @@ executeAndInsertBlockM b = do
       put (rmSetBlockStore s bs')
       ok eb
 
+executeAndInsertBlockE bs0 block =
+  maybeS (getBlock (block ^∙ bId) bs0) continue (pure ∘ (bs0 ,_))
+ where
+  continue : Either ErrLog (BlockStore _ × ExecutedBlock)
+  continue =
+    maybeS (bs0 ^∙ bsRoot _) (Left fakeErr) λ bsr →
```

Comment:
I don't see this in the Haskell code.  Can you make this look more like the Haskell, the Haskell more like this, or if neither, comment on the difference?

Reply:
It is the same.  Have you updated your haskell branch to my latest?

Reply:
Thanks.